src/assignments/benson/lesson_3_answers.clj

```diff
@@ -0,0 +1,128 @@
+(ns assignments.benson.lesson-3-answers)
+
+(declare flush? straight? filter-matching-pairs)
+
+(defn get-suit
+  "Returns the corresponding suit of a card based on the first
+character of PAIR"
+  [pair]
+  (case (first pair)
+    \S :spade
+    \H :heart
+    \D :diamond
+    \C :club))
+
+(defn get-rank
+  "Returns the corresponding rank of a card based on the second
+character of PAIR"
+  [pair]
+  (case (second pair)
+    \2 0
+    \3 1
+    \4 2
+    \5 3
+    \6 4
+    \7 5
+    \8 6
+    \9 7
+    \T 8
+    \J 9
+    \Q 10
+    \K 11
+    \A 12))
+
+(defn derive-pair
+  "Returns a map containing the suit and rank from PAIR"
+  [pair]
+  {:suit (get-suit pair) :rank (get-rank pair)})
+
+; testing assert functionality while doing this exercise
+(assert (= {:suit :diamond :rank 10} (derive-pair "DQ")))
+(assert (= {:suit :heart :rank 3} (derive-pair "H5")))
+(assert (= {:suit :club :rank 12} (derive-pair "CA")))
+(assert (= (range 13) (map (comp :rank derive-pair str)
+                           '[S2 S3 S4 S5 S6 S7
+                             S8 S9 ST SJ SQ SK SA])))
+
+
+(defn straight-flush?
+  "Returns if all pairs in HAND are sequential and are of the same suit"
+  [hand]
+  (and (flush? hand) (straight? hand))
+  )
+
+(defn full-house?
+  "returns a truthy value if a full-house (3 pair + 2 pair) is in HAND
+   otherwise return nil"
+  [hand]
+  (and (filter-matching-pairs hand 3)
```

Comment:
Did you consider re-using pair? and three-of-a-kind?

I.e. build a "tower of abstraction", where find-matching-pairs is all the way at the bottom, and used as little as possible?